### PR TITLE
Minimal of by_gop parallel encoding

### PR DESCRIPTION
**Pull request by lu-zero**
It uses the internals as they are so there is some overhead due unneeded
computation we will remove later.

- [x] properly restrict the pool usage
- [x] part of the channel-api
  - Only single pass rate-control for now.
- [x] rav1e-ch integration

### CODE REVIEW

File: src/api/config/mod.rs [hunk 1]
```diff
@@ -128,6 +128,8 @@ pub struct Config {
   pub(crate) threads: usize,
   /// Shared thread pool
   pub(crate) pool: Option<Arc<ThreadPool>>,
+  /// Number of parallel encoding slots
+  pub(crate) slots: usize,
```
Comment (barrbrain) [inline src/api/config/mod.rs:132]: We can safely excluded this field from stable API builds.
```suggestion
  #[cfg(feature = "unstable")]
  pub(crate) slots: usize,
```

File: src/bin/common.rs [hunk 2]
```diff
@@ -369,6 +370,14 @@ pub fn parse_cli() -> Result<CliOptions, CliError> {
         .help("Overwrite output file.")
         .short("y")
     )
+    .arg(
+      Arg::with_name("SLOTS")
```
Comment (barrbrain) [inline src/bin/common.rs:374]: Perhaps this argument should be hidden for stable builds. I'm not sure how to ergonomically exclude it with this builder syntax.

Comment (lu-zero) [reply src/bin/common.rs:374]: I'll get that sorted.

**Review by barrbrain — APPROVED** [review]
This is a good baseline, contained by `#[cfg(feature = "unstable")]`. Prior to future enhancements, we might want at least to ensure builds with this feature in CI.

**Pull request merged**